### PR TITLE
Implementation of WUDAPT (31-41) landuse classes along with standard urban classes (31-33)

### PR DESCRIPTION
**Pull request by davegill**
TYPE: new feature

KEYWORDS: urban, local climate zone, 

SOURCE: Developer: Andrea Zonato (University of Trento, Italy, andrea.zonato@unitn.it)

DESCRIPTION OF CHANGES:
This update includes the  extension of urban classes from 3 (31-33) to 11 WUDAPT Local Climate zones (31-41). 
See Stewart and Oke, 2012 for further details

Briefly, a new urban parameter table (URBPARAM_LCZ.TBL) has been added for using the new LCZ urban category. 
The new LCZ definition has LU_INDEX=LCZ+30. If a user still wants to keep using the traditional 33 classes 
(31st-33rd for urban), please specify the namelist option "use_wudapt_lcz = 0". If a user wants to use the new LCZ, 
please specify the namelist option "use_wudapt_lcz = 1". If the number of urban category in the input files is 
inconsistent with the namelist option, error messages will occur. Only the LCZ modeling capability is provided here, 
not the dataset which is the responsibility of users. The method to create the LCZ data is described in the World 
Urban Database website: http://www.wudapt.org/

LIST OF MODIFIED FILES:
M Registry/Registry.EM_COMMON
M Registry/Registry.NMM
M main/depend.common
M dyn_em/module_initialize_real.F
M main/depend.common
M phys/Makefile
A phys/module_bep_bem_helper.F
M phys/module_physics_init.F
M phys/module_sf_bep.F
M phys/module_sf_bep_bem.F
M phys/module_sf_clm.F
M phys/module_sf_noahdrv.F
M phys/module_sf_noahlsm.F
M phys/module_sf_noahmpdrv.F
M phys/module_sf_noahmplsm.F
M phys/module_sf_urban.F
M phys/module_surface_driver.F
M run/LANDUSE.TBL
M run/MPTABLE.TBL
M run/URBPARM.TBL
A run/URBPARM_LCZ.TBL
M run/VEGPARM.TBL
A share/module_check_a_mundo.F

TESTS CONDUCTED:
1. Jenkins testing OK

RELEASE NOTE: 
Stewart, I.D. and Oke, T.R. (2012) Local Climate Zones for Urban Temperature Studies. Bulletin of the American Meteorological Society, 93, 1879-1900.
http://dx.doi.org/10.1175/BAMS-D-11-00019.1

### CODE REVIEW

File: dyn_em/module_initialize_real.F [hunk 1]
```diff
@@ -2994,7 +2994,7 @@ SUBROUTINE init_domain_rk ( grid &
             DO i = its , MIN(ide-1,ite)
               IF ( MMINLU == 'NLCD40' .OR. MMINLU == 'MODIFIED_IGBP_MODIS_NOAH') THEN
                   IF ( grid%FRC_URB2D(i,j) .GE. 0.5 .AND.  &
-                      (grid%ivgtyp(i,j).NE.13 .AND. grid%ivgtyp(i,j).NE.24 .AND. grid%ivgtyp(i,j).NE.25 .AND. grid%ivgtyp(i,j).NE.26)) grid%ivgtyp(i,j)=13
+                      (grid%ivgtyp(i,j).NE.13 .AND. grid%ivgtyp(i,j).NE.24 .AND. grid%ivgtyp(i,j).NE.25 .AND. grid%ivgtyp(i,j).NE.26 .AND. grid%ivgtyp(i,j).LT.30)) grid%ivgtyp(i,j)=13
```
Comment (davegill) [inline dyn_em/module_initialize_real.F:2997]: @andreazonato @cenlinhe
Is this required for the LCZ mods?

**Comment by andreazonato** [comment]
Yes, it is, since if not all classes greater than 30 with frc urb less than
0.5 would be turned into class 13

Il ven 26 feb 2021, 21:21 Dave Gill <notifications@github.com> ha scritto:

**Comment by davegill** [comment]
@andreazonato @cenlinhe @weiwangncar @dudhia 
This is a proposed replacement for #1411
This PR does not have the problem of incorrect urban classes based on the use_wudapt_lcz settings, and does not have the resultant slow down for the first time step.

**Comment by davegill** [comment]
jenkins:
```
Please find result of the WRF regression test cases in the attachment. This build is for Commit ID: 01a591bbc0b7068716a693285739d2fe6cdd68a5, requested by: davegill for PR: https://github.com/wrf-model/WRF/pull/1419. For any query please send e-mail to David Gill.

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 19           18
    Number of Builds       : 48           46
    Number of Simulations  : 163           161        0
    Number of Comparisons  : 103           102        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None
```

**Comment by davegill** [comment]
@andreazonato @cenlinhe @dudhia @weiwangncar 
Who would be able to test this PR to see if it actually does the urban classes correctly?

**Comment by cenlinhe** [comment]
@davegill Thanks! I just took a look at your modification in this PR and it looks good to me. I will try to download this PR code and do a quick test to see if the results look reasonable.

**Comment by davegill** [comment]
@cenlinhe 
>  I will try to download this PR code and do a quick test to see if the results look reasonable.

Thanks!

**Comment by andreazonato** [comment]
I read the code, and it seems a reasonable and fast way to do it. Sorry but
I did I with the competences I have,and it was much more complicated and
slow

Il ven 26 feb 2021, 22:19 Dave Gill <notifications@github.com> ha scritto:

**Comment by davegill** [comment]
@andreazonato 
Absolutely no one gets a PR accepted without modifications. No one.

One of the benefits of an open development repository is that everyone can make suggestions to the source code.

That is also the biggest _problem_ with open development. :)

We very much appreciate your scientific modifications to the code, and hope that you continue to be an active WRF model contributor.

**Comment by cenlinhe** [comment]
@davegill @andreazonato I just ran a few tests by using BEP and BEP_BEM scheme with Dave's new updates and it seems that the results are looking reasonable. Here are they: /glade/scratch/cenlinhe/WRF_noahmp_test/WRF_LCZ_test_1419/run/urban2/ and /glade/scratch/cenlinhe/WRF_noahmp_test/WRF_LCZ_test_1419/run/urban3/
I think the LCZ code is ready.

**Review by dudhia — APPROVED** [review]
large number of changes, but tested by LSM group

**Pull request merged**